src/sentry/tasks/reprocessing2.py

```diff
@@ -96,13 +97,19 @@ def reprocess_group(
 
                     continue
 
+            # In case of errors while kicking off reprocessing, mark the event
+            # as reprocessed such that progressbar advances and the
+            # finish_reprocessing task is still correctly spawned.
+            mark_event_reprocessed(group_id=group_id, project_id=project_id)
```

Comment:
Would it make sense to call `mark_event_reprocessed` in a `finally` block here, and remove the call from `_save_event`?